tests/lib/test_io.py

```diff
@@ -1,206 +1,166 @@
 
 
-def test_copy_single_file():
-    fixtures_directory_path = Path(
-        os.path.dirname(__file__), "..", "__fixtures__", "example.yaml"
-    ).resolve()
-    destination_path = Path(
-        os.path.dirname(__file__), "..", "__fixtures__", "example_copy.yaml"
-    ).resolve()
-
-    io.copy(fixtures_directory_path, destination_path)
-
-    assert os.path.isfile(destination_path)
-
-    os.remove(destination_path)
-
-
-def test_copy_directory_with_content():
-    fixtures_directory_path = Path(os.path.dirname(__file__), "..", "__fixtures__").resolve()
-    destination_path = Path(os.path.dirname(__file__), "..", "__fixtures__copy").resolve()
-    fixtures_files = os.listdir(fixtures_directory_path)
-
-    io.copy(fixtures_directory_path, destination_path)
-
-    copied_fixtures_files = os.listdir(destination_path)
-
-    assert os.path.isdir(destination_path)
-    assert fixtures_files == copied_fixtures_files
-
-    shutil.rmtree(destination_path)
-
-
-def test_copy_should_raise_if_failure():
-    with pytest.raises(Exception):
-        fixtures_directory_path = Path(os.path.dirname(__file__), "..", "__abcdefg").resolve()
-        destination_path = Path(os.path.dirname(__file__), "..", "__abcdefg_copy").resolve()
-
-        io.copy(fixtures_directory_path, destination_path)
-
-
-def test_create_temporary_directory():
-    temporary_path = io.create_temporary_directory("prefix")
-
-    assert temporary_path.startswith("/tmp/nestor/work/prefix-")
-    assert os.path.isdir(temporary_path)
-
-    shutil.rmtree(temporary_path)
-
-
-def test_ensure_dir_existing_folder():
-    # "tests" directory of the project
-    tests_directory_path = Path(os.path.dirname(__file__), "..").resolve()
-
-    io.ensure_dir(tests_directory_path)
-
-    assert os.path.isdir(tests_directory_path)
-
-
-def test_ensure_dir_not_existing_folder():
-    # "tests/ensure_dir" directory in the project
-    tests_directory_path = Path(os.path.dirname(__file__), "..", "ensure_dir").resolve()
-
-    assert not os.path.isdir(tests_directory_path)
-
-    io.ensure_dir(tests_directory_path)
-
-    assert os.path.isdir(tests_directory_path)
-
-    os.rmdir(tests_directory_path)
-
-
-def test_execute(mocker):
-    mocker.patch.object(subprocess, "run")
-
-    io.execute("a command with --arg1 arg-value")
-
-    subprocess.run.assert_called_with(
-        ["a", "command", "with", "--arg1", "arg-value"], check=True, cwd=None, stdout=-1
-    )
-
-
-def test_execute_should_raise_if_failure():
-    with pytest.raises(Exception):
+# pylint: disable=no-self-use
+class TestIoLib(TestCase):
+    @patch("nestor_api.lib.io.shutil", autospec=True)
+    def test_copy_single_file(self, shutil_mock):
+        shutil_mock.copytree.side_effect = OSError(errno.ENOTDIR, "some reason")
+        io.copy("source/path/file.yaml", "dest/path/file.yaml")
+        shutil_mock.copytree.assert_called_once()
+        shutil_mock.copy.assert_called_with("source/path/file.yaml", "dest/path/file.yaml")
```

Comment:
I personally think that it is clearer to keep the blank lines between the 3 parts of the test as you did in `test_get_working_path()` (no need for the comments tho). What do you think ?
Same comment for all other tests.
```suggestion
    def test_copy_single_file(self, shutil_mock):
        shutil_mock.copytree.side_effect = OSError(errno.ENOTDIR, "some reason")

        io.copy("source/path/file.yaml", "dest/path/file.yaml")

        shutil_mock.copytree.assert_called_once()
        shutil_mock.copy.assert_called_with("source/path/file.yaml", "dest/path/file.yaml")
```

Reply:
I think it was agreed with @Olivier-Daire and @chris-verclytte to keep the comments just to give it format to the tests. Also I think it will be up to discussion whether we do multiple.patch to avoid cluttering the mock part but I'm no expert here 🙉

Reply:
Yes but for such a small test, I don't think it is necessary